File: Ruby/Saldo/Account.rb
require 'savon'
 
autentica = Savon.client do
    wsdl 'http://pruebascfdi.smartweb.com.mx/Autenticacion/wsAutenticacion.asmx?WSDL'
end
begin
    response = autentica.call(:autenticar_basico) do
        message usuario: 'demo', password: '123456789'
    end
    token = response.body[:autenticar_basico_response][:autenticar_basico_result]
    account = Savon.client do
        wsdl 'https://pruebascfdi.smartweb.com.mx/Account/wsAccount.svc?wsdl'
        ssl_verify_mode :none
        headers 'token' => token
    end
    response = account.call(:get_saldo_client_by_id)
    saldoCliente = response.body[:get_saldo_client_by_id_response][:get_saldo_client_by_id_result]
    puts saldoCliente[:saldo_timbres]
    puts saldoCliente[:timbres_utilizados]
rescue => e
    puts e
end

File: Node.js/Saldo/Account.js
var soap = require('soap');
 
process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
 
function printResult(value) {
    console.log(value);
}
 
function Autentica(args, extractToken) {
    var url = 'http://pruebascfdi.smartweb.com.mx/Autenticacion/wsAutenticacion.asmx?WSDL';
     
    var response = function(err, result) {
        if (err != null)
            throw err.body;
        extractToken(result.AutenticarBasicoResult);
    }
     
    var request = function(err, client) {
        client.AutenticarBasico(args, response);
    }
     
    soap.createClient(url, request);
}
 
function Account(extractResult) {
    var url = 'https://pruebascfdi.smartweb.com.mx/Account/wsAccount.svc?WSDL';
    var header = null;
    var _client = null
     
    var response = function(err, result) {
        if (err != null)
            throw err.body;
        extractResult(result.GetSaldoClientByIdResult.SaldoTimbres);
        extractResult(result.GetSaldoClientByIdResult.TimbresUtilizados);
    }
     
    var request = function(err, client) {
        _client = client;
        client.addSoapHeader(header, 'token', null, 'http://soapinterop.org/echoheader/');
        client.GetSaldoClientById({}, response);
    }
     
    this.init = function(token) {
        header = { token: token };
        soap.createClient(url, request);
    }
}
 
var credentials = { usuario: 'demo', password: '123456789' };
try {
    var account = new Account(printResult);
    var autentica = new Autentica(credentials, account.init);
} catch(err) {
    console.log(err);
}